scripts/lib/CIME/SystemTests/pet.py

```diff
@@ -30,6 +30,10 @@ def _case_one_setup(self):
             if self._case.get_value("NTHRDS_%s"%comp) <= 1:
                 self._case.set_value("NTHRDS_%s"%comp, 2)
 
+        # Subtle: machines like titan will not let you overload the number of threads
```

Comment:
@jedwards4b , I've been thinking and you're right. The real issue is that critical values that are set up in case._initialize_derived_attributes are not updated when case_setup is called, even though the values that went in to compute those derived attributes may have changed. I'm going to test a different fix.